gratipay/utils/team_history.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from datetime import datetime
+from decimal import Decimal as D
+
+
+def get_end_of_year_totals(db, team, year):
+
+    received = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Received
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-team';
+    """, locals() )
+
+    if received is None:
+       received = D(0.00)
+
+    distributed = db.one("""
+        SELECT COALESCE(sum(amount), 0) AS Distributed
+          FROM payments
+         WHERE team = %(team)s
+           AND extract(year from timestamp) = %(year)s
+           AND amount > 0
+           AND direction='to-participant';
+    """, locals())
+
+    if distributed is None:
+        distributed = D(0.00)
+
+    return received, distributed
+
+
+def iter_team_payday_events(db, team, year=None):
```

Comment:
I don't think we should keep `year` as an optional here, it already defaults to `current_year` in the template ( `year = int(request.qs.get('year', current_year))` ). 

Not that this is wrong, just that it is one more thing to maintain :)